app/View/Delete.js

```diff
@@ -1,21 +1,104 @@
 import React from 'react';
-import Inflector from 'inflected'
+import Inflector from 'inflected';
+import {Link} from 'react-router';
+import {shouldComponentUpdate} from 'react-immutable-render-mixin';
+
 import ViewActions from '../Component/ViewActions';
 import Compile from '../Component/Compile';
+import EntityActions from '../Actions/EntityActions';
+import EntityStore from '../Stores/EntityStore';
 
 class DeleteView extends React.Component {
+    constructor() {
+        super();
+
+        this.shouldComponentUpdate = shouldComponentUpdate.bind(this);
+    }
+
+    componentDidMount() {
+        EntityStore.addDeleteListener(this.onDelete.bind(this));
+        EntityStore.addChangeListener(this.onChange.bind(this));
+
+        this.refreshData();
+    }
+
+    componentWillUnmount() {
+        EntityStore.removeChangeListener(this.onDelete.bind(this));
+    }
+
+    onChange() {
+        this.setState(EntityStore.getState());
+    }
+
+    refreshData() {
+        let {id} = this.context.router.getCurrentParams();
+
+        EntityActions.loadEditData(this.props.configuration, this.getView(), id);
```

Comment:
You should load specific data for delete, for example we don't need values from `ReferencedList`.